### PR TITLE
Muq 34

### PR DESCRIPTION
**Pull request by mnzit**

### CODE REVIEW

File: src/main/java/life/genny/qwandaq/utils/minio/Minio.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            InputStream object = minioClient.getObject(GennySettings.getMinIOBucketName(),
+                    userUUID.toString() + "/media/" + fileUUID.toString());
+            byte[] byteArray = IOUtils.toByteArray(object);
+            return byteArray;
+        } catch (InvalidKeyException | InvalidBucketNameException
+                 | NoSuchAlgorithmException | InsufficientDataException
+                 | NoResponseException | ErrorResponseException
+                 | InternalException | InvalidArgumentException | IOException
+                 | XmlPullParserException e) {
+            e.printStackTrace();
```
Comment (varunshastry) [inline src/main/java/life/genny/qwandaq/utils/minio/Minio.java]: Add log statement with exception details

File: src/main/java/life/genny/qwandaq/utils/minio/Minio.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    public boolean uploadFile(String sub, String inpt, String uuid) {
+        boolean isSuccess = false;
+
+        String path = sub + "/" + "media" + "/" + uuid;
+        try {
+            boolean isExist = minioClient.bucketExists(GennySettings.getMinIOBucketName());
+            if (isExist) {
+                System.out.println("Bucket " + GennySettings.getMinIOBucketName() + "already exists.");
+            } else {
+                System.out.println("Start creat Bucket:" + GennySettings.getMinIOBucketName());
```
Comment (varunshastry) [inline src/main/java/life/genny/qwandaq/utils/minio/Minio.java]: System.out.println needs to be replaced with "log"ger statements

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: src/main/java/life/genny/qwandaq/models/GennySettings.java [hunk 3]
```diff
@@ -90,4 +89,29 @@ public static String keycloakUrl() {
 			: "http://keycloak.genny.life";
 	}
 
+	//MINIO
+	public static String getMinIOServerUrl(){
+		return System.getenv("MINIO_SERVER_URL") != null
+				? System.getenv("MINIO_SERVER_URL")
```
Comment (varunshastry) [inline src/main/java/life/genny/qwandaq/models/GennySettings.java]: Extract to "Constants" file

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: src/main/java/life/genny/qwandaq/models/GennySettings.java [hunk 4]
```diff
@@ -90,4 +89,29 @@ public static String keycloakUrl() {
 			: "http://keycloak.genny.life";
 	}
 
+	//MINIO
+	public static String getMinIOServerUrl(){
+		return System.getenv("MINIO_SERVER_URL") != null
```
Comment (varunshastry) [inline src/main/java/life/genny/qwandaq/models/GennySettings.java]: Extract to "Constants". Also, store value of System.getenv() into a variable, and return that variable if not null. That would avoid 2 function calls.

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: src/main/java/life/genny/qwandaq/models/GennySettings.java (hunk 4 above)
Comment (mnzit) [reply src/main/java/life/genny/qwandaq/models/GennySettings.java]: /**
	 * NOTE: The variables above seem to be defaulting all the time.
	 * likely due to an issue with the environment variables not being present at initialisation.
	 * If it cannot be fixed, we may have to opt for doing it as seen below.
	 **/
Storing the variable to constant gave me issue in runtime so using getters

**Pull request merged**